ci/build_python.sh

```diff
@@ -6,20 +6,21 @@ set -euo pipefail
 source rapids-env-update
 
 CONDA_CONFIG_FILE="conda/recipes/versions.yaml"
+export CONDA_OVERRIDE_CUDA="${RAPIDS_CUDA_VERSION}"
```

Comment:
We should probably consider the implications here. Today, I think rapids is installable even on CPU-only machines. The new rapids-xgboost package design requires `__cuda` to **install**. This is important to support for cases like HPC systems with CPU login nodes and GPU worker nodes that use the same environment.

Reply:
Yeah they can also install by setting `CONDA_OVERRIDE_CUDA` to some value

In any event, this is [coming from `libxgboost`]( https://github.com/conda-forge/xgboost-feedstock/blob/829bd2ea851c5b4eddafad846c79e333835a45cb/recipe/meta.yaml#L61 ). So we could move this just to the `rapids-xgboost` if we prefer

Reply:
> The new rapids-xgboost package design 

What is the relevant new change? Is it in libxgboost or in something about how rapids-xgboost is packaged?

Reply:
Okay. Well, maybe this requirement already existed. I'm not sure.

The question is whether `__cuda` should be a hard requirement for installation, which is coming from xgboost-related packages. I'm not sure if it was that way for the old xgboost packages we shipped in 23.06 or not. Regardless, it feels funny that no other RAPIDS package has this requirement besides xgboost.

Reply:
Dropping this in PR ( https://github.com/rapidsai/integration/pull/673 ), which pulls in the new `xgboost` packages